################################################################################
#
# Title:        ONTAP-12-03 - Local User Access
# Author:       NetApp Inc. (badrian)
# Initial 
# Create Date:  2023-08-23
# Description:  Cluster User Management
#               - Local User Access
#
# URLs:         https://docs.netapp.com/us-en/ontap/index.html
#               https://galaxy.ansible.com/netapp/ontap
#
# Built-in help:
# ansible-doc netapp.ontap.<module_name>
# 
################################################################################

- hosts:              "ontap"
  name:               "ONTAP-12-03 - Local User Access"
  gather_facts:       false
  vars:
    input_ro: &input_ro
      hostname:                   "{{ ansible_host }}"
      username:                   "{{ ontap_12_local_ro }}"
      password:                   "{{ vault_security_accounts[ontap_12_local_ro].password }}"
      https:                      true
      validate_certs:             false
      use_rest:                   "always"
    input_admin: &input_admin
      hostname:                   "{{ ansible_host }}"
      username:                   "{{ ontap_12_local_admin }}"
      password:                   "{{ vault_security_accounts[ontap_12_local_admin].password }}"
      https:                      true
      validate_certs:             false
      use_rest:                   "always"
  vars_files:
    - ../vars/labondemand/vars.yml
    - ../vars/labondemand/vault.yml
  collections:
    - netapp.ontap

  pre_tasks:

  tasks:
    - name: Creating a test SVM with different accounts having different privileges
      block:
        - name: Create test SVM with read-only account
          netapp.ontap.na_ontap_svm:
            state:                      present
            name:                       "{{ ontap_12_test_svm01 }}"
            <<: *input_ro
          when:
            - ontap_12_test_svm01 is defined
          register: ro_create_result

      rescue:
        - name: Printing rescue message - Expected error
          ansible.builtin.debug:
            msg:
              - "This error was expected"
              - "RO user {{ input_ro.username }} cannot create ressources on the cluster"
          when:
            - ("not authorized for that command") in ro_create_result.msg

        - name: Printing rescue message - Unexpected error
          ansible.builtin.debug:
            msg:
              - "Unexpected test result:"
              - "{{ ro_create_result }}"
          when:
            - not ("not authorized for that command") in ro_create_result.msg

        - name: Create test SVM with admin account
          netapp.ontap.na_ontap_svm:
            state:                      present
            name:                       "{{ ontap_12_test_svm01 }}"
            <<: *input_admin
          when:
            - ontap_12_test_svm01 is defined

      - name: Clean-up test SVM with admin account
        netapp.ontap.na_ontap_svm:
          state:                      absent
          name:                       "{{ ontap_12_test_svm01 }}"
          <<: *input_admin
        when:
          - ontap_12_test_svm01 is defined
  
  post_tasks: